primus_vk.cpp

```diff
@@ -140,14 +170,42 @@ VK_LAYER_EXPORT VkResult VKAPI_CALL PrimusVK_CreateInstance(
     VkPhysicalDeviceProperties props;
     dispatchTable.GetPhysicalDeviceProperties(dev, &props);
     TRACE(GetKey(dev) << ": ");
-    if(props.deviceType == VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU){
-      display = dev;
-      TRACE("got display!");
+    if(displayVendorID != 0){
+      if(props.vendorID == displayVendorID){
+        if(props.deviceID == displayDeviceID){
+          display = dev;
+          TRACE("got display from PRIMUS_VK_DISPLAYID!");
+        }
+        else {
+          display = dev;
+          TRACE("got display from PRIMUS_VK_DISPLAYID (via vendorID)!");
+        }
+      }
+    } else {
+      if(props.deviceType == VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU){
+        display = dev;
+        TRACE("got integrated display!");
+      }
```

Comment:
maybe this whole decision logic is complex enough that it could be beneficial to extract it into a new method.

Reply:
It should be more robust now in this latest commit, and separated display & render gpu for loops.